elliottlib/cli/find_builds_cli.py

```diff
@@ -127,103 +133,78 @@ def find_builds_cli(runtime, advisory, default_advisory_type, builds, kind, from
     # Returns a list of (n, v, r, pv) tuples of each build
-    image_tuples = []
     component_names = []
-    latest_builds = {}
     for i in image_metadata:
         component_names.append(i.get_component_name())
 
-    for tag in tag_pv_map:
-        latest_builds = brew.get_latest_builds(tag, component_names)
-        for _, b in latest_builds.items():
-            image_tuples.append((b['name'], b['version'], b['release'], tag_pv_map[tag]))
+    image_tuples = _gen_nvrp_tuples('name', component_names, tag_pv_map)
 
     pbar_header(
         'Generating build metadata: ',
         'Fetching data for {n} builds '.format(n=len(image_tuples)),
         image_tuples)
 
-    # By 'meta' I mean the lil bits of meta data given back from
-    # get_latest_build_info
-    #
-    # TODO: Update the ImageMetaData class to include the NVR as
-    # an object attribute.
-    results = parallel_results_with_progress(
-        image_tuples,
-        lambda meta: elliottlib.brew.get_brew_build(
-            '{}-{}-{}'.format(meta[0], meta[1], meta[2]),
-            product_version=meta[3],
-            session=session)
-    )
-
-    return [
-        b for b in results
-        if not b.attached_to_open_erratum
-        # filter out 'openshift-enterprise-base-container' since it's not needed in advisory
-        if 'openshift-enterprise-base-container' not in b.nvr
-    ]
+    return image_tuples
 
 
-def _fetch_builds_by_kind_rpm(builds, base_tag, product_version, session):
+def _fetch_builds_by_kind_rpm(tag_pv_map):
     green_prefix('Generating list of rpms: ')
     click.echo('Hold on a moment, fetching Brew builds')
-    candidates = elliottlib.brew.find_unshipped_build_candidates(
-        base_tag,
-        product_version,
-        kind='rpm')
-
-    pbar_header('Gathering additional information: ', 'Brew buildinfo is required to continue', candidates)
-    # We could easily be making scores of requests, one for each build
-    # we need information about. May as well do it in parallel.
-    results = parallel_results_with_progress(
-        candidates,
-        lambda nvr: elliottlib.brew.get_brew_build(nvr, product_version, session=session)
-    )
-    return _attached_to_open_erratum_with_correct_product_version(results, product_version, elliottlib.errata)
-
-
-def _attached_to_open_erratum_with_correct_product_version(results, product_version, errata):
+    rpm_tuple = []
+    for tag in tag_pv_map:
+        if tag.endswith('-candidate'):
+            base_tag = tag[:-10]
+        else:
+            red_print("key of brew_tag_product_version_mapping in erratatool.yml must be candidate\n")
+            continue
+        candidates = elliottlib.brew.find_unshipped_build_candidates(
+            base_tag,
+            tag_pv_map[tag],
+            kind='rpm')
+        pbar_header('Gathering additional information: ', 'Brew buildinfo is required to continue', candidates)
+        rpm_tuple.extend(_gen_nvrp_tuples('nvr', candidates, tag_pv_map))
```

Comment:
`find_unshipped_build_candidates` calls brew list-tagged CLI then `_gen_nvrp_tuples` also calls Brew APIs. Those Brew CLI/API calls can be combined for better performance but can be done in separated cards.